txstratum/jobs.py

```diff
@@ -0,0 +1,171 @@
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_header_without_nonce(self) -> bytes:
+        """Return job's header without nonce."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_nonce_size(self) -> int:
+        """Return job's nonce size."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_data(self) -> bytes:
+        """Return data to be submitted to the backend."""
+        raise NotImplementedError
+
+
+class MinerTxJob(MinerJob):
+    """Tx job."""
+
+    def __init__(self, data: bytes):
+        """Init TxJob."""
+        self.data: bytes = data
+
+        self._tx: Transaction = Transaction.create_from_struct(data)
+
+        self.uuid: bytes = self.get_uuid(data)
+        self.is_block: bool = False
+        self.share_weight: float = 0
+        self.status: JobStatus = JobStatus.PENDING
+        self.created_at: float = time.time()
+        self.submitted_at: Optional[float] = None
+        self.total_time: Optional[float] = None
+        self.nonce: Optional[bytes] = None
+
+    @classmethod
+    def get_uuid(cls, tx: bytes) -> bytes:
+        """Return the job uuid."""
+        s1 = hashlib.sha256(tx).digest()
+        s2 = hashlib.sha256(s1).digest()
+        return s2
+
+    def get_data(self) -> bytes:
+        """Return data to be submitted to the backend."""
+        return self.data
+
+    def mark_as_solved(self, nonce: bytes) -> None:
+        """Mark job as solved."""
+        now = time.time()
+        self.status = JobStatus.DONE
+        self.nonce = nonce
+        self.submitted_at = now
+        self.total_time = now - self.created_at
+
+    def get_header_without_nonce(self) -> bytes:
+        """Return job's header without nonce."""
+        return self._tx.get_header_without_nonce()
+
+    def get_nonce_size(self) -> int:
+        """Return job's nonce size."""
+        return self._tx.SERIALIZATION_NONCE_SIZE
+
+    def get_weight(self) -> float:
+        """Return job's weight (difficulty)."""
+        return self._tx.weight
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Return a dict with an overview of the job.
+
+        Returns:
+            job_id: str, job identifier
+            status: str, choices: pending, mining, done, failed, cancelled
+            created_at: int, timestamp that the job was submitted
+            expected_queue_time: int, expected time in queue (in seconds)
+            expected_mining_time: int, expected time to be mined (in seconds)
+            expected_total_time: int, sum of expected_queue_time and expected_mining_time (in seconds)
+        """
+        return {
+            'job_id': self.uuid.hex(),
+            'status': self.status.value,
+            'created_at': self.created_at,
+            'nonce': self.nonce.hex() if self.nonce else None,
+            'submitted_at': self.submitted_at,
+            'total_time': self.total_time,
+            # TODO
```

Comment:
Important TODO because the estimation values are still hardcoded

Reply:
Done. Thanks!